### PR TITLE
MoveFile moves existing file to root of project when renaming them.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @balajikris

File: src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.RenameFile.cs [hunk 1]
```diff
@@ -34,6 +34,28 @@ class Inner { }
         }
 
         [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public async Task TestRenameFileWithFolders()
+        {
+            var code =
+@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document Folders=""A\B"">
```
Comment (balajikris) [inline src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.RenameFile.cs]: nit: if Document tag supports giving it a name, specifying the name here makes test clearer (file name <> type name)

File: src/EditorFeatures/Test/MoveType/AbstractMoveTypeTest.cs [hunk 2]
```diff
@@ -96,6 +97,12 @@ protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspa
                     // the original source document does not exist in the new solution.
                     var newSolution = oldSolutionAndNewSolution.Item2;
                     Assert.Null(newSolution.GetDocument(oldDocumentId));
+
+                    var newDocument = newSolution.Projects.Single().Documents.Single();
```
Comment (balajikris) [inline src/EditorFeatures/Test/MoveType/AbstractMoveTypeTest.cs]: is there a better way to get the document form the new solution than single?

File: src/EditorFeatures/Test/MoveType/AbstractMoveTypeTest.cs [hunk 3]
```diff
@@ -96,6 +97,12 @@ protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspa
                     // the original source document does not exist in the new solution.
                     var newSolution = oldSolutionAndNewSolution.Item2;
                     Assert.Null(newSolution.GetDocument(oldDocumentId));
+
```
Comment (balajikris) [inline src/EditorFeatures/Test/MoveType/AbstractMoveTypeTest.cs]: Looks like we're not asserting that the file rename actually happened :( . It should, given that this is `TestRenameFileToMatchTypeAsync`. If its simple, could you add an assert for that -- Thanks!

File: src/EditorFeatures/Test/MoveType/AbstractMoveTypeTest.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/MoveType/AbstractMoveTypeTest.cs]: Not that i know of.

**Comment by CyrusNajmabadi** [comment]
retest this please

**Pull request merged**